dashboard/app/controllers/admin_reports_controller.rb

```diff
@@ -0,0 +1,169 @@
+# The controller for reports of internal admin-only data.
+class AdminReportsController < ApplicationController
+  before_filter :authenticate_user!, except: [:header_stats]
+
+  before_action :set_script
```

Comment:
(Note that before_action is just a more modern synonym for before_filter.  However, I don't object to using before_filter for hooks that might reject the request and before_action for things that set some controller state.)